tests/load/sources/filesystem/test_filesystem_source.py

```diff
@@ -0,0 +1,273 @@
+            csv_duckdb_reader.with_name("csv_duckdb_example"),
+        ]
+    )
+    # pandas incorrectly guesses that taxi dataset has headers so it skips one row
+    # so we have 1 less row in csv_example than in csv_duckdb_example
+    assert_load_info(load_info)
+    assert load_table_counts(
+        pipeline,
+        "jsonl_example",
+        "parquet_example",
+        "listing",
+        "csv_example",
+        "csv_duckdb_example",
+    ) == {
+        "jsonl_example": 1034,
+        "parquet_example": 1034,
+        "listing": 11,
+        "csv_example": 1279,
+        "csv_duckdb_example": 1280,
+    }
+    # print(pipeline.last_trace.last_normalize_info)
+    # print(pipeline.default_schema.to_pretty_yaml())
+
+
+@pytest.mark.parametrize("bucket_url", TESTS_BUCKET_URLS)
+def test_incremental_load(bucket_url: str) -> None:
+    @dlt.transformer
+    def bypass(items) -> str:
+        return items
+
+    pipeline = dlt.pipeline(
+        pipeline_name="file_data",
+        destination="duckdb",
+        dataset_name="filesystem_data_duckdb",
+        full_refresh=True,
+    )
+
+    # Load all files
+    all_files = filesystem(bucket_url=bucket_url, file_glob="csv/*")
+    # add incremental on modification time
+    all_files.apply_hints(incremental=dlt.sources.incremental("modification_date"))
+    load_info = pipeline.run((all_files | bypass).with_name("csv_files"))
+    assert_load_info(load_info)
+    assert pipeline.last_trace.last_normalize_info.row_counts["csv_files"] == 4
+
+    table_counts = load_table_counts(pipeline, "csv_files")
+    assert table_counts["csv_files"] == 4
+
+    # load again
+    all_files = filesystem(bucket_url=bucket_url, file_glob="csv/*")
+    all_files.apply_hints(incremental=dlt.sources.incremental("modification_date"))
+    load_info = pipeline.run((all_files | bypass).with_name("csv_files"))
+    # nothing into csv_files
+    assert "csv_files" not in pipeline.last_trace.last_normalize_info.row_counts
+    table_counts = load_table_counts(pipeline, "csv_files")
+    assert table_counts["csv_files"] == 4
+
+    # load again into different table
+    all_files = filesystem(bucket_url=bucket_url, file_glob="csv/*")
+    all_files.apply_hints(incremental=dlt.sources.incremental("modification_date"))
+    load_info = pipeline.run((all_files | bypass).with_name("csv_files_2"))
+    assert_load_info(load_info)
+    assert pipeline.last_trace.last_normalize_info.row_counts["csv_files_2"] == 4
+
+
+def test_file_chunking() -> None:
+    resource = filesystem(
+        bucket_url=TESTS_BUCKET_URLS[0],
+        file_glob="*/*.csv",
+        files_per_page=2,
+    )
+
+    from dlt.extract.pipe_iterator import PipeIterator
+
+    # use pipe iterator to get items as they go through pipe
+    for pipe_item in PipeIterator.from_pipe(resource._pipe):
+        assert len(pipe_item.item) == 2
+        # no need to test more chunks
+        break
+
+
+@pytest.mark.parametrize(
```

Comment:
with this setup you'll always run the example pipeline against duckdb, not against the currently active destination. I'm not quite sure how to inject the right destination for this test, maybe in the future we can make use of the updated dlt init command that is run before the tests to create a file with the right destination from which we can import this, not sure.

 I'd also suggest at least doing a very basic inspection of the loaded data by attaching to the pipeline, but maybe this is not needed..